lib/src/native/realm_core.dart

```diff
@@ -448,6 +451,65 @@ class _RealmCore {
     return RealmHandle._(realmPtr);
   }
 
+  RealmSchema readSchema(Realm realm) {
+    return using((Arena arena) {
+      final actualCount = arena<Size>();
+      Pointer<Uint32> classesPtr;
+      var classesSize = 10;
+
+      while (true) {
```

Comment:
I feel bad we are introducing these endless loops in the code. Why not call this function again. I think I had a comment about this somewhere. No every other function is using an endless loop. This is error prone and could lead to problems if the code is refactored. Also its hard to read and reason about. Its much more clear if it calls itself with the correct size again.

Reply:
I know that it's rare that a schema changes between two invocations of some code, but I wanted to strictly adhere to the Core API and make sure we don't introduce some very rare crashes where a user randomly gets an empty array because we provided a container that's too small.

Reply:
that's more readable code right there. It reads better, it is not hardcoded endless recursion and can not unintentionally become one if one refactors the code later on. The `while(true)` is just asking for trouble and is really hard to read imo. I hope we don't start using goto labels :)

Also there is no need to `arena.free(classesPtr);` in the `if` explicitly since this will get called automatically when the arena is torn down. Yes we will keep that allocation a bit longer but I think this is the idea of the arena to not bother freeing the memory. But freeing will also work.

Reply:
That code is perfect ^

Reply:
btw you can drop the `Internal`. `_readSchema` can co exists with `readSchema`